test/Dialect/Substrait/literal.mlir

```diff
@@ -1,6 +1,29 @@
 // RUN: substrait-opt -split-input-file %s \
 // RUN: | FileCheck %s
 
+// CHECK:      substrait.plan version 0 : 42 : 1 {
+// CHECK-NEXT:   relation
+// CHECK:         %[[V0:.*]] = named_table
+// CHECK-NEXT:    %[[V1:.*]] = project %[[V0]] : tuple<si1> -> tuple<si1, !substrait.uuid> {
+// CHECK-NEXT:    ^[[BB0:.*]](%[[ARG0:.*]]: tuple<si1>):
+// CHECK-NEXT:      %[[V2:.*]] = literal "c48ffa9e-64f4-44cb-ae47-152b4e60e77b" : !substrait.uuid
+// CHECK-NEXT:      yield %[[V2]] : !substrait.uuid
+// CHECK-NEXT:    }
+// CHECK-NEXT:    yield %[[V1]] : tuple<si1, !substrait.uuid> 
+
+substrait.plan version 0 : 42 : 1 {
+  relation {
+    %0 = named_table @t1 as ["a"] : tuple<si1>
+    %1 = project %0 : tuple<si1> -> tuple<si1, !substrait.uuid> {
+    ^bb0(%arg : tuple<si1>):
+      %uuid = literal "c48ffa9e-64f4-44cb-ae47-152b4e60e77b" : !substrait.uuid
```

Comment:
This is obviously much more readable than a 16-byte string. However, we can get the best of both world (readability and easy conversion from protobuf+more memory-efficient representation) by defining proper `UUIDAttr` with a custom assembly format that prints/parses to/from hex/binary. That could be a TODO if we don't want to spend time on that detail now.